simpleviewinc/keg-cli
# Move mutagen to repo

## Context
* In a effort to breakdown the keg-cli into smaller modules, we are migrating modules to their own repo folder
 * The code in the `/src/libs` folder will be migrated to the repos folder each as it's own repo

### Updates
**cli-utils** - minor version update
* Update the `GLOBAL_CONFIG_FOLDER` and `GLOBAL_CONFIG_FILE` to be exported from the constants

**mutagen-lib**
* Migrate `src/libs/mutagen` to `repos/mutagen-lib`
  * Made mutagen-lib it's own package 
  * Updated dependencies to pull from `cli-utils`
  * Removed all keg-cli dependencies

**keg-cli**
* Updated `KegMutagen` alias to use the `mutagen-lib` package
* Update url for keg-cli


### Test

**Setup**
* Pull the pr => `keg cli && keg pr <>`
* Run yarn install => `yarn install`
* Because the changes to cli-utils has not been published yet, we need to update the `package.json` of `mutagen-lib` to use the local version
  * Update the `dependencies` => `@keg-hub/cli-utils` to look like this => `"@keg-hub/cli-utils": "file:../cli-utils",`
    * Then navigate to the `mutagen-lib` folder and run `yarn install`

**Test 1**
* Run cli tests => `keg cli && yarn test`
* Run cli-utils tests => `keg cli && cd repos/cli-utils && yarn test`
* Run mutagen tests => `keg cli && cd repos/mutagen-lib && yarn test`

**Test 2**
* Run a tap with a mutagen sync => `keg evf start`
* Make a change to a file
  * Ensure the changes are sync into the running container 

**Test 3**
* Run a sync action => `keg evf sync components:install:start`
* Make a change to keg-components
  * Ensure the changes are sync'ed into the container  
* Open another terminal window and run `keg sy`
  * Ensure all mutagen syncs are listed by running `keg sy` in a di

**Test 4**
* Kill the running tap => `keg evf kill`
* Open another terminal window and run `keg sy`
  * Ensure all syncs for the tap are removed
